### PR TITLE
GhiaC

### PR DESCRIPTION
**Pull request by GhiaC**

### CODE REVIEW

**Comment by 1995parham** [comment]
First of all, I want to thank you for your solution to this project. I think your code is good but let me test it and I will let you know as soon as I can.

File: main.go [hunk 1]
```diff
@@ -1,7 +1,99 @@
...
-	fmt.Println("18.20 at Sep 07 2016 7:20 IR721")
+	configuration := conf.Configuration{}
+	_ = gonfig.GetConf("conf.json", &configuration)
+
+	ctx := context.Background() ///WithTimeout(context.Background(), 10*time.Second)
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI("mongodb://localhost:27017"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = client.Ping(ctx, readpref.Primary())
```
Comment (1995parham) [inline main.go]: I did not see this earlier. can you explain it to me?

File: main.go [hunk 2]
```diff
@@ -1,7 +1,99 @@
...
+		}
+	})
+	e.GET("/getAll", func(c echo.Context) error {
+		cur, err := collection.Find(ctx, bson.D{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		var arr []models.Location
+		for cur.Next(ctx) {
+			var result bson.M
```
Comment (1995parham) [inline main.go]: You can use a specific structure for this instead of bson.M.

File: main.go [hunk 3]
```diff
@@ -1,7 +1,99 @@
 package main
 
-import "fmt"
+import (
+	"./conf"
```
Comment (1995parham) [inline main.go]: use absolute package address `${module_name}/conf` for example: `github.com/aolab/letsgo/conf`

**Review by 1995parham — COMMENTED** [review]
Again I want to thank you for your time and contribution. 💪

File: main.go (hunk 3 above)
Comment (GhiaC) [reply main.go]: done

**Comment by GhiaC** [comment]
> Again I want to thank you for your time and contribution. muscle

Thanks for doing this and it's my pleasure

File: main.go (hunk 2 above)
Comment (GhiaC) [reply main.go]: done

File: main.go (hunk 1 above)
Comment (GhiaC) [reply main.go]: Ping verifies that the client can connect to the topology. If readPreference is nil then will use the client's default read preference.
https://godoc.org/go.mongodb.org/mongo-driver/mongo#Client.Ping

**Pull request merged**